src/hotspot/cpu/arm/c1_LIRGenerator_arm.cpp

```diff
@@ -285,10 +285,11 @@ LIR_Address* LIRGenerator::emit_array_address(LIR_Opr array_opr, LIR_Opr index_o
 }
 
 
-LIR_Opr LIRGenerator::load_immediate(int x, BasicType type) {
+LIR_Opr LIRGenerator::load_immediate(jlong x, BasicType type) {
   assert(type == T_LONG || type == T_INT, "should be");
   LIR_Opr r = make_constant(type, x);
-  bool imm_in_range = AsmOperand::is_rotated_imm(x);
+  bool imm_in_range = (x >= 0) && (x <= UINT_MAX) &&
```

Comment:
The check is too strict, what about x >= INT_MIN && x <= INT_MAX?

Reply:
How is it strict? We are about to cast to _unsigned int_, surely we want to test that the value is in range of unsigned int?

Reply:
The method is correct but a bit unoptimal. I may be wrong, but _x_ is signed variable and may contain negative value that fits _logic_and_ (like ~0xFF00)

Reply:
All right, let me just blindly cast to "unsigned int" then. If this is the problem in ARM32, it can be fixed by follow-up bug. See new commit.